docs/libraries/core.md

```diff
@@ -302,9 +302,40 @@ Less than or equal. This component is combinational.
 
 ## Memories
 
+Calyx features two flavors of memories: combinational and sequential.
+Combinational memories promise that they will return `mem[addr]` in the same cycle that `addr` is provided.
+Sequential memories, on the other hand, promise that they will return `mem[addr]` in the next cycle after `addr` is provided.
+We generally encourage the use of sequential memories as they are more realistic.
+Combinational memories are useful when the memory is known to be small, and the application is very performance-sensitive.
+
+### `seq_mem_d1`
+
+A one-dimensional memory with sequential reads.
+
+**Parameters:**
+
+- `WIDTH` - Size of an individual memory slot.
+- `SIZE` - Number of slots in the memory.
+- `IDX_SIZE` - The width of the index given to the memory.
+
+**Inputs:**
+
+- `addr0: IDX_SIZE` - The index to be accessed or updated.
+- `write_data: WIDTH` - Data to be written to the selected memory slot.
+- `write_en: 1` - One bit write enabled signal. Used in concert with `content_en`; see below.
+- `content_en: 1` - One bit content enabled signal. When `content_en` is high and `write_en` is low, the memory reads the value stored at `addr0` and latches it. When `write_en` and `content_en` are both high, the memory writes `write_data` to the slot indexed by `addr0` and sets the latched output to `'x`.
```

Comment:
Maybe instead of saying "and sets the latched output to `'x`," it would be marginally more helpful to say "and `read_data` is undefined"? That is, `'x` isn't exactly a Calyx-level feature; we can just say that people should not pay attention to the `read_data` signal in this case.

Reply:
Super, thanks!